File: PointFreeLeetCode/PointFree.LeetCode.Tests_XUnit/Problem_905.fs
module PointFree.LeetCode.Tests_XUnit.Problem_905

open FsCheck.FSharp
open FsCheck.Xunit
open PointFree.LeetCode_CSharp
open PointFree.LeetCode

[<Property>]
let ``Implementation is correct`` (array: int array) =
    let s = Problem_905()
   
    s.SortArrayByParity array = Problem_905.sortArrayByParity array

File: PointFreeLeetCode/PointFree.LeetCode_CSharp/Problem_905.cs
namespace PointFree.LeetCode_CSharp;

public class Problem_905
{
    public int[] SortArrayByParity(int[] nums)
    {
        var i = 0;
        var j = nums.Length - 1;
        for (; i <= j; i++)
        {
            if (nums[i] % 2 == 0) continue;
            
            for (; j > i; j--)
            {
                if (nums[j] % 2 == 0)
                {
                    (nums[j], nums[i]) = (nums[i], nums[j]);
                    break;
                }
            }
        }
        
        return nums;
    }
}